### DISCUSSION TITLE
Idea: Being able to download all projects on your profile

### DISCUSSION THREAD

**Discussion by weeniemount** [ScratchAddons/ScratchAddons#5744] [Ideas]
It would be cool if you could download every project that you have in your project folder. For example, moving projects easily to difffrent accounts, backing up projects, etc...

**Comment by DNin01** [comment]
For people with hundreds or even thousands of projects, this would be very time-, resource-, and storage-consuming. I'm not sure if downloading all of them at a time would be feasible.

An alternative would be to add a Download button on each project in the My Stuff list.

But the real solution is to make a habit of downloading your projects to your computer every time you save new changes so you don't have to download them all at once.

**Comment by Samq64** [comment]
The only way this would really work is if all the projects are downloaded as one zip, but the file would be huge depending on the number of projects the user has.

**Comment by WorldLanguages** [comment]
This would only make sense for mass-downloading own unshared projects.
A tool that does this with someone's public projects could exist already, if someone built it.

**Comment by mybearworld** [comment]
This might also spam the Scratch API too much.

**Comment by cheekyrandom** [comment]
Could be risky, because it **may and could utilize something** that has the Scratch user's token.
In all honesty, it can be done. and as @mybearworld said,

*This might also spam the Scratch API too much.*

So we're just stuck to downloading them manually.

<h5>p.s it would need to be disguised as a friendly program that is a trojan to not make it obvious that it's a cookie stealer and a trojan<h5>

<h4>GOSH I HATE EDITS<h4>

**Reply by mybearworld** [reply]
Why would it be a trojan?

**Reply by Secret-chest** [reply]
Scratch says 10 requests per second is OK. That'd be max. 5 projects / second (we need two requests for each). Still much faster than doing it all by hand.

**Reply by DNin01** [reply]
> Scratch says 10 requests per second is OK. That'd be max. 5 projects / second (we need two requests for each). Still much faster than doing it all by hand.

What about all the assets of each project?

**Reply by Secret-chest** [reply]
> > Scratch says 10 requests per second is OK. That'd be max. 5 projects / second (we need two requests for each). Still much faster than doing it all by hand.
> 
> What about all the assets of each project?

Still much faster than doing it by hand.

**Comment by DNin01** [comment]
Could we allow individually downloading projects instead of downloading all of them at once?

**Reply by WorldLanguages** [reply]
I encourage you to create a new issue with the "new addon" template: "download project on My Stuff"
But I'm not sure about the technical aspects of downloading a project outside of the editor. We might need to ship a lot of code as well as a zip library.
After we figure that out we can decide whether we can let the user download more than one project at once as a zip file. Or possibly download many individual files at once (but browsers don't allow that by default)

**Reply by DNin01** [reply]
Created issue: #5968
